src/modules/Checkbox/Checkbox.js

```diff
@@ -107,57 +108,51 @@ export default class Checkbox extends Component {
 
   handleClick = (e) => {
     debug('handleClick()')
+    debug('firing change event')
+
     const { disabled, onChange, onClick, name, readOnly, value } = this.props
-    // using a ref here allows us to let the browser manage radio group state for us
-    // this is a special exception where we are reading state from the DOM
-    // otherwise, all radio groups would have to be controlled components
-    const refChecked = _.get(this.refs, 'input.checked')
+    const { checked } = this.state
```

Comment:
Did away with refs for Checkboxes.  This is one more antipattern off the list.  Ideally, we never read from the real DOM but store all state in the virtual DOM only.